src/Components/Login.js

```diff
@@ -0,0 +1,69 @@
+import React, { Component } from 'react';
+
+const uri = process.env.REACT_APP_API_ENDPOINT || 'http://localhost:8080'
+
+class Login extends Component {
+
+  state = {
+    username: '',
+    player: null
+  }
+
+  componentDidMount() {
+    this.loginStatus();
+  }
+
+  loginStatus() {
+    return fetch(`${uri}/login`, {
+      method: 'GET',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      credentials: 'include'
+    }).then(res => {
+      if(res.status < 400) return res.json()
+      else return null
+    }).then(user => {
+      this.setState({...this.state, player: user})
```

Comment:
I don't think you need to have the `...this.state` when calling this.setState(), it should but able to take `{player: user}` and update the player field, without modifying anything else (in this case, username).

Reply:
you are correct, they are merged.